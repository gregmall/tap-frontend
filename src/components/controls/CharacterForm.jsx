import React from 'react';
import PropTypes from 'prop-types';
import styles from './CharacterForm.css';

const CharacterForm = ({
  name,
  image,
  quote,
  role,
  onChange,
  
}) =>{
    return (

      <div className={styles.CharacterForm}>
        <section>
          <label htmlFor="name">Name:</label>
          <input id="name"
            type="text"
            name="name"
            value={name}
            placeHolder={name}
            onChange={onChange} /><br />
          <label htmlFor="image">Image Url:</label>
          <input id="image"
            type="text"
            name="image" value={image}
            placeHolder={image}
            onChange={onChange} /><br />
          <textarea
            placeHolder="Enter quote here"
            id="quote"
            type="textarea"
            name="quote"
            value={quote}
            placeHolder={quote}
            onChange={onChange} /><br />
          <label htmlFor="role">Role:</label>
          <input id="role"
            type="text"
            name="role" value={role}
            placeHolder={role}
            onChange={onChange} />
        </section>
      </div>


    );
  };

CharacterForm.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  quote: PropTypes.string.isRequired,
  role: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
  
};

export default CharacterForm;

