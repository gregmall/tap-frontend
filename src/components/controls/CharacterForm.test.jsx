import React from 'react';
import { render } from '@testing-library/react';
import CharacterForm from './CharacterForm';

describe ('WordForm controller component', () => {
  it('should render a form to add a character to the screen', () => {
    const { asFragment } = render(
      <CharacterForm 
        name="Steve Johnson"
        image="www.rockandroll.com"
        quote="live and die for music"
        role="roadie"
        onChange={() => { }}

      />)
      
      expect(asFragment()).toMatchSnapshot()

  })

})
