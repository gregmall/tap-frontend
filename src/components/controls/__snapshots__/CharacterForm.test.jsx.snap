// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WordForm controller component should render a form to add a character to the screen 1`] = `
<DocumentFragment>
  <div
    class="CharacterForm"
  >
    <section>
      <label
        for="name"
      >
        Name:
      </label>
      <input
        id="name"
        name="name"
        placeholder="Steve Johnson"
        type="text"
        value="Steve Johnson"
      />
      <br />
      <label
        for="image"
      >
        Image Url:
      </label>
      <input
        id="image"
        name="image"
        placeholder="www.rockandroll.com"
        type="text"
        value="www.rockandroll.com"
      />
      <br />
      <textarea
        id="quote"
        name="quote"
        placeholder="live and die for music"
        type="textarea"
      >
        live and die for music
      </textarea>
      <br />
      <label
        for="role"
      >
        Role:
      </label>
      <input
        id="role"
        name="role"
        placeholder="roadie"
        type="text"
        value="roadie"
      />
    </section>
  </div>
</DocumentFragment>
`;
